llvm/lib/Target/AIE/AIE2PreLegalizerCombiner.cpp

```diff
@@ -136,13 +138,34 @@ bool AIE2PreLegalizerCombinerImpl::tryToCombineVectorShiftsByZero(
   return true;
 }
 
+bool AIE2PreLegalizerCombinerImpl::tryToCombineSetExtract(
```

Comment:
Can you add tests?

Reply:
FYI @andcarminati maybe you're also introducing something similar?

Reply:
We have some new ones on Post Custom Combiner now, more related to vector updates, so it is different ;-).